src/fides/api/service/saas_request/override_implementations/oracle_responsys_request_overrides.py

```diff
@@ -136,42 +136,46 @@ def oracle_responsys_profile_list_recipients_read(
     else:
         list_ids = list_ids_from_api
 
-    identity = get_identity(privacy_request)
-    if identity == "email":
-        query_ids = input_data.get("email", [])
-        query_attribute = "e"
-    elif identity == "phone_number":
-        query_ids = input_data.get("phone_number", [])
-        # Oracle Responsys doesn't support the initial + in phone numbers
-        for idx, query_id in enumerate(query_ids):
-            query_ids[idx] = query_id[1:] if query_id.startswith("+") else query_id
-        query_attribute = "m"
-    else:
-        raise FidesopsException(
-            "Unsupported identity type for Oracle Responsys connector. Currently only `email` and `phone_number` are supported"
-        )
+    for identity in get_identities(privacy_request):
+        if identity == "email":
```

Comment:
We could extract the code fragment for retrieving the query_ids and query_attributes into its own method for readability, now that we are one level deeper with a for loop.   It would help readability